<launch>
  <arg name="test" default="1"/>
  <arg name="run" default="4"/>

  <arg name="bag_in" value = "/media/justin/data/real_testing/test_$(arg test)/test_$(arg run).bag"/>
  <arg name="record_out" value= "/media/justin/data/real_testing/rtabmap_records/test_$(arg test)_test_$(arg run).db"/>

  <include file="$(find rtabmap_ros)/launch/data_recorder.launch">
    <arg name="subscribe_odometry" value="true"/>
    <arg name="frame_id" value="base_footprint"/>
    <arg name="rgb_topic" value="/camera/rgb/image_rect_color"/>
    <arg name="depth_topic" value="/camera/depth_registered/sw_registered/image_rect"/>
    <arg name="rgb_info_topic" value="/camera/rgb/camera_info"/>
    
    <arg name="output_path" value="$(arg record_out)"/>
    <arg name="record_in_RAM" value="true"/>
    <arg name="queue_size" value="1000"/>


  </include>  


  <include file="$(find tb)/launch/rosbag_play_turtlebot.launch" >
    <arg name="rviz" value="false"/>
  </include>


  <node name="bag_player" pkg="rosbag" type="play" args="$(arg bag_in) --clock --pause --skip-empty=1 --queue=1000 -r 1  ">
    <remap from="/camera/depth/image" to="/camera/depth/image_raw"/>
    <remap from="/camera/rgb/image_color" to="/camera/rgb/image_raw"/>
  </node>
  

</launch>
